runtime/nativeMain/src/kotlinx/benchmark/NativeBlackhole.kt

```diff
@@ -1,35 +1,61 @@
 package kotlinx.benchmark
 
-import kotlinx.cinterop.pin
+import kotlinx.cinterop.convert
+import kotlinx.cinterop.toByte
+import kotlin.concurrent.Volatile
+import kotlin.native.ref.WeakReference
+import kotlin.random.Random
+import kotlinx.benchmark.impl.*
 
+@OptIn(ExperimentalStdlibApi::class)
 actual class Blackhole {
+    /*
+     * For primitive types, DCE may be omitted by creating an artificial use via inline asm.
+     * For objects, there is no cheap way to obtain something that could be passed to a native function with such
```

Comment:
Worth asking `@Pavel Kunyavskiy` about this one.
AFAIR, there was _some_ way (kinda internal-ish) to cast `Any` to `long*`, and then we can do the same trick as with primitives (first ensuring that it does prevent e.g. scalarization)

Reply:
Well, there's not dedicated API to do that, but there's `identityHashCode` (and I totally forgot about its existence) which is effectively `reinterpret_cast<uintptr_t>(this)`.